### ISSUE TITLE
(Unmanaged) Strings in callbacks are not considered as blittable by WasmAppBuilder

### ISSUE THREAD

**Issue by aienabled** [dotnet/runtime#92380]
### Description

The strings are not considered as blittable when defined in Pinvoke callbacks.
https://github.com/dotnet/runtime/blob/a5b75b8179d7db4ff720e19233d2b99fd15b94f0/src/tasks/WasmAppBuilder/PInvokeCollector.cs#L172

### Reproduction Steps

```cs
private delegate void Callback_RegisterType(string typeName);
private static Callback_RegisterType _registerType = RegisterType;

[MonoPInvokeCallback(typeof(Callback_RegisterType))]
private static void RegisterType(string typeName)
{
}
```

### Expected behavior

The above code should compile.

### Actual behavior

WasmApp.Native.targets(290,5): Error : Parameter types of pinvoke callback method 'System.Void RegisterType(System.String)' needs to be blittable.

### Configuration

Tested on .NET 7.0.11. "WebAssembly Browser App" project type.

### Other information

I'm working on porting a UI library to WASM and its C# SDK makes extensive use of strings in Pinvoke callbacks (dozens of methods). The C# SDK has no such error when building for other platforms (including WebGL in Unity).

**Comment by radical** [comment]
cc @maraf

**Comment by Lurler** [comment]
Any news about this? This is extremely important for us.

**Comment by kg** [comment]
Strings aren't blittable, see https://learn.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types

I assume what you want is for strings to be accepted as arguments to pinvoke callbacks.

Are you unable to pass char pointers instead by pinning your strings? (I'm aware that doing this would be an annoying change to make, so I'm not discounting the value of adding support for strings as arguments)

**Comment by dp901** [comment]
This is an issue that has been blocking us for a while. Is there any chance it could be addressed sooner?

**Comment by aienabled** [comment]
@kg
> Strings aren't blittable

Indeed, but there is no problem using the code as shown in the snippet above on other platforms. I suppose the strings are automatically converted to char pointers.

> I assume what you want is for strings to be accepted as arguments to pinvoke callbacks.

Yes, exactly.

> Are you unable to pass char pointers instead by pinning your strings? (I'm aware that doing this would be an annoying change to make, so I'm not discounting the value of adding support for strings as arguments)

That's a good idea though it would be quite cumbersome to maintain a fork considering there are dozens of methods where a string is passed in a callback. However, when I attempted to rewrite the code to use char pointers, I encountered another compilation problem (the delegate/callback pointer declared with [UnmanagedFunctionPointer(CallingConvention.Cdecl)] is also not blittable, again this is not an issue on other platforms). I suppose we will have to wait until .NET WASM becomes more mature.

**Comment by Thraka** [comment]
@kg 9 is getting close! :)
